src/Dev/Tasks/TagsToShortcodeHelper.php

```diff
@@ -0,0 +1,290 @@
+        if (strpos($tag, $needle)) {
+            $propertyValue = explode('"', explode($needle, $tag)[1])[0];
+        }
+
+        return $propertyValue;
+    }
+
+    /**
+     * Extracts an array of attributes from a tag.
+     * @param string $tag
+     * @param string $tagType
+     * @param $id
+     * @param $newSrc
+     * @return array
+     */
+    private static function getAttributes(string $tag, string $tagType, $newSrc)
+    {
+        $attributes = [];
+
+        if ($tagType == 'img') {
+            $imgAttributes = [
+                'src' => $newSrc,
+            ];
+
+            if ($title = static::getAttribute($tag, 'title')) {
+                $imgAttributes['title'] = $title;
+            }
+            if ($width = static::getAttribute($tag, 'width')) {
+                $imgAttributes['width'] = $width;
+            }
+            if ($height = static::getAttribute($tag, 'height')) {
+                $imgAttributes['height'] = $height;
+            }
+            if ($class = static::getAttribute($tag, 'class')) {
+                $imgAttributes['class'] = $class;
+            }
+            if ($alt = static::getAttribute($tag, 'alt')) {
+                $imgAttributes['alt'] = $alt;
+            }
+
+            $attributes = array_merge($attributes, $imgAttributes);
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * @param string $tagType
+     * @return string
+     */
+    private static function getShortCodeTypeFromTagType(string $tagType)
+    {
+        return static::VALID_TAGS[$tagType];
+    }
+
+    private function getParsedFileIDFromSrc(string $src)
+    {
+        $fileIDHelperResolutionStrategy = new FileIDHelperResolutionStrategy();
+        $fileIDHelperResolutionStrategy->setResolutionFileIDHelpers([
+            new HashFileIDHelper(),
+            new LegacyFileIDHelper(),
+            $defaultFileIDHelper = new NaturalFileIDHelper(),
+        ]);
+        $fileIDHelperResolutionStrategy->setDefaultFileIDHelper($defaultFileIDHelper);
+
+        $filesystem = $this->flysystemAssetStore->getPublicFilesystem();
+
+        // set fileID to the filepath relative to assets dir
+        $pattern = '/^\\' . DIRECTORY_SEPARATOR . '?' . ASSETS_DIR . '?\\' . DIRECTORY_SEPARATOR . '?/';
+        $fileID = preg_replace($pattern, '', $src);
+
+        return $fileIDHelperResolutionStrategy->softResolveFileID($fileID, $filesystem);
+    }
+
+    /**
+     * @param string $tag
+     * @return string|null Returns the new tag or null if the tag does not need to be rewritten
+     */
+    private function getNewTag(string $tag)
+    {
+        list('tagType' => $tagType, 'src' => $src) = $this->getTagTuple($tag);
+
+        // Search for a File object containing this filename
+        $parsedFileID = $this->getParsedFileIDFromSrc($src);
+        /** @var File $file */
+        if ($parsedFileID && ($file = File::get()->filter('FileFilename:PartialMatch', $parsedFileID->getFilename())->first())) {
```

Comment:
Why are you using a partial match here? Files are normally identified by their FileFilename and FileHash combination.

Also you may need to check for live files as well.

Reply:
How would you do that here using the ORM?